source/core/database/mssql.ts

```diff
@@ -0,0 +1,479 @@
+     */
+    public constructor(options: MSSQLDatabaseOptions) {
+        super();
+        this._databaseName = options.databaseName;
+        this._db = knex({
+            client: 'mssql',
+            debug: options.debug ?? true,
+            acquireConnectionTimeout: options.connectionTimeout ?? 20000,
+            log: {
+                debug: (debug): void => {
+                    if (options.debug) {
+                        const tables = this._extractTablesFromSqlQuery(debug.sql as string);
+                        const eventDebugLog: MssqlEventLog = {
+                            database: options.databaseName,
+                            tables,
+                            queryUuid: debug.__knexQueryUid,
+                            method: debug.method,
+                            sql: debug.sql,
+                            bindings: debug.bindings
+                        };
+                        this.emit(EVENT_MSSQL.LOG, eventDebugLog);
+                    }
+                }
+            },
+            connection: {
+                database: options.databaseName,
+                host: options.host,
+                port: options.port,
+                user: options.user,
+                password: options.password,
+                options: {
+                    encrypt: options.encrypt ?? true
+                },
+                bigNumberStrings: false
+            },
+            pool: {
+                min: options.poolMin ?? 2,
+                max: options.poolMax ?? 10
+            }
+        });
+    }
+
+    /**
+     * Establishes a connection to the MSSQL database.
+     *
+     * If the connection is successful, introspection is performed to retrieve information about tables, columns(fields), and primary keys.
+     *
+     * @throws ({@link CoreError}) Thrown if an error occurs during the connection process. ({@link DATABASE_KEY_ERROR.MSSQL_CONNECTION_ERROR})
+     */
+    public async connect(): Promise<void> {
+        try {
+            await this._introspectDatabase();
+            this._isConnected = true;
+            this._addEventKnex();
+        } catch (error) {
+            throw new CoreError({
+                key: DATABASE_KEY_ERROR.MSSQL_CONNECTION_ERROR,
+                cause: error
+            });
+        }
+    }
+
+    /**
+     * Closes the connection to the MSSQL database.
+     *
+     * @throws ({@link CoreError}) Thrown if the database is not connected. ({@link DATABASE_KEY_ERROR.MSSQL_NOT_CONNECTED})
+     * @throws ({@link CoreError}) Thrown if an error occurs during the disconnection process. ({@link DATABASE_KEY_ERROR.MSSQL_DISCONNECT_ERROR})
+     */
+    public async disconnect(): Promise<void> {
+        if (!this._isConnected)
+            throw new CoreError({
+                key: DATABASE_KEY_ERROR.MSSQL_NOT_CONNECTED
+            });
+        try {
+            await this._db.destroy();
```

Comment:
Lorque la connexion se ferme ne devrait-on pas supprimer tous les listeners pour éviter que les objets Table ou autres objets liés aux événements ne restent en mémoire
```suggestion
            this.removeAllListeners()
```